fix(gltf): Preserve accessor properties in Draco decoded mesh attrributes

Avnerus opened the pull request:
Hi,
This one is a bit more tricky as I am not sure if this was the best way to address my problem:
When loading glTFs (from Cesium 3d-tiles) into three.js, the `min`, `max` attributes of the mesh attribute accessor were lost, which [prevented three.js](https://github.com/mrdoob/three.js/blob/master/examples/jsm/loaders/GLTFLoader.js#L2944-L2945) from computing a bounding box for the mesh. What I did in this PR was preserve the original json accessor of the draco decoded attribute, and then assign any missing properties to that attribute.
I also had to modify one test, making sure I delete the resolved accessor min/max values so that the gltf builder would calculate them on its own (the floating point values ended up being just a bit different).
I think overall this could improve performance because it saves the need to calculate min/max values for draco decoded meshes. Let me know what you think!

/Avner

ibgreen commented:
@Avnerus Thanks for reporting, I am looking at this in the background to consider the best approach here. In the mean-time, @dryabinin-actionengine  could potentially merge this patch into `2.3-release` branch and we can do a more solid overhaul in 3.0.

ibgreen commented:
2.3.12

Avnerus commented:
@ibgreen , @dryabinin-actionengine . Now updated for `master`. Would it be possible to get this into 3.0?
Thank you!

/Avner

Avnerus commented:
Closing since this is now merged into the Typescript refactor. 
Thank you!

Avnerus commented:
@ibgreen 
Rebased PR and added some types, hopefully the change is now clearer.
Also reverted one change in the `parse` call which prevented the worker from being used.

ibgreen commented on modules/gltf/src/lib/extensions/KHR_draco_mesh_compression.ts:
```diff
@@ -76,9 +81,24 @@ async function decompressPrimitive(
   const dracoOptions = {...options};
```
Yes we need to do it this way. 

Unfortunately, going through `parse()` defeats the new typings. Not sure if that is fixable. We could explicitly import types from the draco loader here.
```suggestion
import {DracoLoaderOptions, DracoMeshData} from `@loaders.gl/draco';
  const dracoOptions: DracoLoaderOptions = {...options};
```

ibgreen commented on modules/gltf/src/lib/extensions/KHR_draco_mesh_compression.ts:
```diff
@@ -76,9 +81,24 @@ async function decompressPrimitive(
   // TODO - remove hack: The entire tileset might be included, too expensive to serialize
   delete dracoOptions['3d-tiles'];
-  const decodedData = await DracoLoader.parse(bufferCopy, dracoOptions, context);
+  const decodedData = await parse(bufferCopy, DracoLoader, dracoOptions, context);
```
Especially having the type checks on the returned data would give me a lot of piece of mind. That was kind of the whole point of the big exercise I made.
```suggestion
  const decodedData = await parse(bufferCopy, DracoLoader, dracoOptions, context) as DracoMeshData;
```

ibgreen commented on modules/gltf/test/lib/api/gltf-scenegraph-modifiers.spec.js:
```diff
@@ -61,6 +61,9 @@ test('GLTFScenegraph#Should calculate min and max arrays for accessor', async (t
   const inputData = await load(GLTF_BINARY_URL, GLTFLoader, {gltf: {postProcess: true}});
   const gltfBuilder = new GLTFScenegraph();
 
+  delete inputData.meshes[0].primitives[0].attributes.POSITION.min;
```
Add comment
```suggestion
  // addMesh does not yet support adding additional accessor attributes, and does not auto calculate them
  delete inputData.meshes[0].primitives[0].attributes.POSITION.min;
```

ibgreen reviewed (approved):
Good to land, thanks!
